### DISCUSSION TITLE
Removing old JFR recordings with LOOP option

### DISCUSSION THREAD

**Discussion by petrbouda** [async-profiler/async-profiler#897] [General]
Hi all,

is there any option to keep only limited number of files with loop option?
I would like to have 1-day profiles and keep only 2 of them + the current that is being appended (recording.jfr.1~).

Is there any way to achieve this, or any best practice ? Before I start with a custom clean-up solution based on bash :)

Thanks a lot,
Petr

**Comment by apangin** [comment]
Yes, you can use `%n` pattern in the file name to limit the number of recordings.
For example, `-f profile-%n{3}.jfr` will write into 3 files in a loop:
  - profile-0.jfr
  - profile-1.jfr
  - profile-2.jfr
  - profile-0.jfr
  - profile-1.jfr
  ...

See #549 for details.

Alternatively, setup a [cron job](https://askubuntu.com/questions/789602/auto-delete-files-older-than-7-days) to remove files basing on their age.

**Reply by petrbouda** [reply]
The pattern-style looks sufficient enough :)
Thanks a lot, Andrei
